patterns/1-initial/modular-code.md

```diff
@@ -8,54 +8,69 @@ The lack of modularization in the software architecture prevents reuseability, a
 
 ## Problem
 
-Engineering does not want to spend the extra resources needed to develop modular components. This applies to managers and developers alike. As a result, fewer components end up being reused, leading to duplication of work.
+* With rare exceptions, monolithinc code is difficult to contribute to.
+* The same functionality is reimplemented redundantly causing
+  * Waste of unneeded re-design and re-development
+  * Duplicated mainteinance effort
```

Comment:
```suggestion
With rare exceptions, monolithic code is difficult to contribute to and extend.
That leads development teams to build similar functionality multiple times, leading to both waste (unneeded re-design and re-development) as well as duplicated maintenance effort.
```

Reply:
In terms of formatting, I am suggesting to not use bullets here. We already have enough bullets in the other sections :)

Also fixed some typos and rewrote the Problem statement a bit.

Reply:
Typos and formatting: ok

I see that _monolithic code_ might be a cause for _redundant implementation_, but it is not always so.  Redundant implementation might happen for other reasons, so I'd keep it as another problem that modularization solves.

Reply:
Sorry but I could not follow your train of thought here, please help me :)

Are you saying that your lines 11 and 12 above are describing two distinct unrelated problems, that can be improved through more modular code?

Reply:
Yes, 2 different problems:

1. Monolithic code packing much functionality makes it difficult to contribute (much to read, much to grasp, much to test, much to review, and much to document). Finding slots and the right mood to attack big chunks of work is difficult. This is regardless of how many other alternative implementations you have.
2. Redundant implementations tend to be a waste of effort developing, documenting, and maintaining them all.

Strictly speaking, modularity is about reusing parts (modules) and solves redundancy.

But engineering the parts to be small and narrow-focused (breaking the monolith) helps a lot in fostering collaboration. More so in the case of volunteer collaboration.